### PR TITLE
[Peft] fix saving / loading when unet is not "unet"

### PR DESCRIPTION
**Pull request by kashif**
# What does this PR do?

Fix lora load and save when the unet is not the "unet" I believe this is a regression from #5991

### CODE REVIEW

File: src/diffusers/loaders/lora.py [hunk 1]
```diff
@@ -1376,10 +1376,10 @@ def pack_weights(layers, prefix):
             )
 
         if unet_lora_layers:
-            state_dict.update(pack_weights(unet_lora_layers, "unet"))
+            state_dict.update(pack_weights(unet_lora_layers, cls.unet_name))
 
         if text_encoder_lora_layers and text_encoder_2_lora_layers:
-            state_dict.update(pack_weights(text_encoder_lora_layers, "text_encoder"))
+            state_dict.update(pack_weights(text_encoder_lora_layers, cls.text_encoder_name))
             state_dict.update(pack_weights(text_encoder_2_lora_layers, "text_encoder_2"))
```
Comment (sayakpaul) [inline src/diffusers/loaders/lora.py:1383]:
```suggestion
            state_dict.update(pack_weights(text_encoder_2_lora_layers, f"{cls.text_encoder}_2"))
```

**Comment by kashif** [comment]
so the error happens when I load lora weights for the Wuerschten lora trainer that if you recall has no `unet` but rather a `self.prior` unet:

```
  File "/mnt/scratch/kashif/diffusers/examples/wuerstchen/text_to_image/train_text_to_image_lora_prior.py", line 894, in main
    pipeline.prior_pipe.load_lora_weights(args.output_dir, weight_name="pytorch_lora_weights.safetensors")
  File "/mnt/scratch/kashif/diffusers/src/diffusers/loaders/lora.py", line 114, in load_lora_weights
    self.load_lora_into_unet(
  File "/mnt/scratch/kashif/diffusers/src/diffusers/loaders/lora.py", line 449, in load_lora_into_unet
    inject_adapter_in_model(lora_config, unet, adapter_name=adapter_name)
  File "/mnt/scratch/kashif/peft/src/peft/mapping.py", line 163, in inject_adapter_in_model
    peft_model = tuner_cls(model, peft_config, adapter_name=adapter_name)
  File "/mnt/scratch/kashif/peft/src/peft/tuners/lora/model.py", line 119, in __init__
    super().__init__(model, config, adapter_name)
  File "/mnt/scratch/kashif/peft/src/peft/tuners/tuners_utils.py", line 95, in __init__
    self.inject_adapter(self.model, adapter_name)
  File "/mnt/scratch/kashif/peft/src/peft/tuners/tuners_utils.py", line 255, in inject_adapter
    raise ValueError(
ValueError: Target modules {'unet.blocks.2.attention.to_k', 'unet.blocks.50.attention.to_v', 'unet.blocks.26.attention.to_k', 'unet.blocks.68.attention.to_v', 'unet.blocks.44.attention.to_k', 'unet.blocks.38.attention.to_q', 'unet.blocks.56.attention.to_k', 'unet.blocks.35.attention.to_v', 'unet.blocks.89.attention.to_k', 'unet.blocks.14.attention.to_v', 'unet.blocks.23.attention.to_q', 'unet.blocks.53.attention.to_out.0', 'unet.blocks.35.attention.to_out.0', 'unet.blocks.2.attention.to_q', 'unet.blocks.44.attention.to_q', 'unet.blocks.74.attention.to_out.0', 'unet.blocks.17.attention.to_out.0', 'unet.blocks.83.attention.to_k', 'unet.blocks.89.attention.to_q', 'unet.blocks.32.attention.to_k', 'unet.blocks.47.attention.to_k', 'unet.blocks.26.attention.to_v', 'unet.blocks.8.attention.to_k', 'unet.blocks.92.attention.to_k', 'unet.blocks.20.attention.to_out.0', 'unet.blocks.56.attention.to_q', 'unet.blocks.83.attention.to_q', 'unet.blocks.41.attention.to_k', 'unet.blocks.20.attention.to_k', 'unet.blocks.86.attention.to_out.0', 'unet.blocks.71.attention.to_out.0', 'unet.blocks.77.attention.to_v', 'unet.blocks.59.attention.to_q', 'unet.blocks.47.attention.to_q', 'unet.blocks.5.attention.to_out.0', 'unet.blocks.35.attention.to_k', 'unet.blocks.71.attention.to_v', 'unet.blocks.65.attention.to_k', 'unet.blocks.29.attention.to_out.0', 'unet.blocks.62.attention.to_out.0', 'unet.blocks.11.attention.to_q', 'unet.blocks.23.attention.to_out.0', 'unet.blocks.74.attention.to_q', 'unet.blocks.50.attention.to_q', 'unet.blocks.29.attention.to_k', 'unet.blocks.17.attention.to_v', 'unet.blocks.26.attention.to_q', 'unet.blocks.53.attention.to_v', 'unet.blocks.77.attention.to_k', 'unet.blocks.23.attention.to_k', 'unet.blocks.44.attention.to_v', 'unet.blocks.29.attention.to_q', 'unet.blocks.59.attention.to_out.0', 'unet.blocks.23.attention.to_v', 'unet.blocks.35.attention.to_q', 'unet.blocks.14.attention.to_out.0', 'unet.blocks.80.attention.to_out.0', 'unet.blocks.11.attention.to_v', 'unet.blocks.89.attention.to_out.0', 'unet.blocks.26.attention.to_out.0', 'unet.blocks.47.attention.to_v', 'unet.blocks.92.attention.to_v', 'unet.blocks.17.attention.to_q', 'unet.blocks.74.attention.to_v', 'unet.blocks.59.attention.to_k', 'unet.blocks.11.attention.to_out.0', 'unet.blocks.44.attention.to_out.0', 'unet.blocks.80.attention.to_q', 'unet.blocks.68.attention.to_q', 'unet.blocks.83.attention.to_v', 'unet.blocks.71.attention.to_q', 'unet.blocks.32.attention.to_q', 'unet.blocks.5.attention.to_v', 'unet.blocks.8.attention.to_out.0', 'unet.blocks.56.attention.to_out.0', 'unet.blocks.38.attention.to_out.0', 'unet.blocks.53.attention.to_q', 'unet.blocks.5.attention.to_k', 'unet.blocks.92.attention.to_out.0', 'unet.blocks.29.attention.to_v', 'unet.blocks.68.attention.to_out.0', 'unet.blocks.74.attention.to_k', 'unet.blocks.65.attention.to_v', 'unet.blocks.41.attention.to_q', 'unet.blocks.68.attention.to_k', 'unet.blocks.41.attention.to_v', 'unet.blocks.95.attention.to_out.0', 'unet.blocks.14.attention.to_q', 'unet.blocks.62.attention.to_v', 'unet.blocks.2.attention.to_out.0', 'unet.blocks.17.attention.to_k', 'unet.blocks.83.attention.to_out.0', 'unet.blocks.80.attention.to_v', 'unet.blocks.65.attention.to_q', 'unet.blocks.86.attention.to_v', 'unet.blocks.32.attention.to_out.0', 'unet.blocks.80.attention.to_k', 'unet.blocks.62.attention.to_k', 'unet.blocks.38.attention.to_k', 'unet.blocks.50.attention.to_k', 'unet.blocks.59.attention.to_v', 'unet.blocks.8.attention.to_v', 'unet.blocks.50.attention.to_out.0', 'unet.blocks.11.attention.to_k', 'unet.blocks.86.attention.to_q', 'unet.blocks.86.attention.to_k', 'unet.blocks.77.attention.to_q', 'unet.blocks.77.attention.to_out.0', 'unet.blocks.32.attention.to_v', 'unet.blocks.89.attention.to_v', 'unet.blocks.20.attention.to_v', 'unet.blocks.62.attention.to_q', 'unet.blocks.92.attention.to_q', 'unet.blocks.71.attention.to_k', 'unet.blocks.95.attention.to_v', 'unet.blocks.53.attention.to_k', 'unet.blocks.95.attention.to_q', 'unet.blocks.41.attention.to_out.0', 'unet.blocks.5.attention.to_q', 'unet.blocks.65.attention.to_out.0', 'unet.blocks.38.attention.to_v', 'unet.blocks.20.attention.to_q', 'unet.blocks.47.attention.to_out.0', 'unet.blocks.14.attention.to_k', 'unet.blocks.8.attention.to_q', 'unet.blocks.2.attention.to_v', 'unet.blocks.56.attention.to_v', 'unet.blocks.95.attention.to_k'} not found in the base model. Please check the target modules and try again.
```

**Review by sayakpaul — APPROVED** [review]
Thank you!

Could you also run the PEFT LoRA slow tests just to make sure things are not being broken here?

**Review by younesbelkada — APPROVED** [review]
Looks great! Can you also run the slow tests as suggested by sayak?

**Comment by kashif** [comment]
sure i'll test with slow tests

**Comment by younesbelkada** [comment]
note you might want to run them on a V100

**Comment by sayakpaul** [comment]
@kashif a gentle ping.

**Pull request merged**